tests/trio103.py

```diff
@@ -0,0 +1,90 @@
+import trio
+
+try:
+    pass
+except (SyntaxError, ValueError, BaseException):
+    raise
+except (SyntaxError, ValueError, trio.Cancelled) as p:  # error
+    pass
+except (SyntaxError, ValueError):
+    raise
+except trio.Cancelled as e:
+    raise e
+except:
+    pass
+
+try:
+    pass
+except BaseException:
+    raise ValueError()  # error
+
+
+def foo(var):
+    try:
+        pass
+    except trio.Cancelled:  # error
+        pass
+    except BaseException as e:  # error
+        if True:
+            raise e
+        if True:
+            pass
+        else:
+            raise e
+
+    try:
+        pass
+    except BaseException as e:
+        try:
+            raise e
+        except trio.Cancelled as f:
+            raise f  # ???
```

Comment:
This is weird but OK.

Reply:
I kinda wanna print that out and put on the wall, I get that feeling so much trying to come up with corner cases - though a lot of them only look weird because the test cases are so stripped down. Somewhere out there there's a good case for writing deeply nested `except`'s

Quoted line 41 won't be considered an error, but I don't consider the parent exception covered.